ext/cl_khr_extended_versioning.asciidoc

```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+[[cl_khr_extended_versioning]]
+== Extended versioning
+
+This extension introduces new platform and device queries that return detailed
+version information to applications. It makes it possible to return the exact
+revision of the specification or intermediate languages supported by an
+implementation. It also enables implementations to communicate a version
+number for each of the extensions they support and remove the requirement
+for applications to process strings to test for the presence of an extension or
+intermediate language or built-in kernel.
+
+=== General information
+
+==== Name Strings
+
+`cl_khr_extended_versioning`
+
+==== Contributors
+
+Kévin Petit, Arm Ltd. +
+Ben Ashbaugh, Intel +
+Alastair Murray, Codeplay Software Ltd. +
+Einar Hov, Arm Ltd.
+
+==== Version history
+
+[cols="1,1,3",options="header",]
+|====
+| *Date*     | *Version* | *Description*
+| 2020-02-12 | 1.0.0     | Initial version.
+|====
+
+==== Dependencies
+
+This extension is written against the OpenCL Specification
+Version 2.2, Revision 11.
+
+This extension requires OpenCL 1.0.
+
+=== New API Types
+
+==== Version
+
+This extension introduces a new scheme to encode detailed
+(major, minor, patch/revision) version information into a single 32-bit unsigned
+integer:
+
+* The major version is using bits 31-22
+* The minor version is using bits 21-12
+* The patch version is using bits 11-0
+
+This scheme enables two versions to be ordered using the standard C/C++ operators.
+Macros are provided to extract individual fields or compose a full version
+from the individual fields.
+
+[source,c]
+----
+
+typedef cl_uint cl_version_khr;
+
+#define CL_VERSION_MAJOR_BITS_KHR (10)
+#define CL_VERSION_MINOR_BITS_KHR (10)
+#define CL_VERSION_PATCH_BITS_KHR (12)
```

Comment:
For the main specification these values are in the XML as constants, this could be too.

Reply:
Yup, done.